### DISCUSSION TITLE
[Suggestion]about NVM_SYMLINK

### DISCUSSION THREAD

**Discussion by SetsuikiHyoryu** [coreybutler/nvm-windows#992] [Ideas & Feature Requests] [answered]
I suggest adding a note in the documentation that specifies the `NVM_SYMLINK` path needs to be manually created if it doesn't exist. The program doesn't create it automatically (other programs usually create the path specified by the user during the installation process).

I encountered an issue where I successfully installed Node using `nvm install`, but after running `nvm use` and then `nvm current`, it displayed a message saying Node was not set. I spent a lot of time troubleshooting and finally realized that it was because the path specified by `NVM_SYMLINK` had not been automatically created.

**Comment by coreybutler** [comment]
`NVM_SYMLINK` **_is_** created during the installation process. Did you use the official installer?

Run `nvm debug` to track down common `PATH` issues.

**Reply by SetsuikiHyoryu** [reply]
Thank you for your answer. I am using version `1.1.11` of `nvm-setup.exe` downloaded from this repository. However, my `NVM_SYMLINK` is not using the default `C:\Program Files\Nodejs`.

Here is the result of my `nvm debug` command. It was displaying the same (just no node.js version information) before I created folder manually. I also tried administrative rights.

```shell
❯ nvm debug
Hyoryu is not using admin or elevated rights.

PowerShell 7.3.2
Windows Developer Mode: UNKNOWN (user cannot read registry)

NVM4W Version:      1.1.11
NVM4W Path:         E:\SOFT\programming\package-manage\nvm\nvm.exe
NVM4W Settings:     E:\SOFT\programming\package-manage\nvm\settings.txt
NVM_HOME:           E:\SOFT\programming\package-manage\nvm
NVM_SYMLINK:        E:\SOFT\programming\programming-language\javascript\nodejs
Node Installations: E:\SOFT\programming\package-manage\nvm

Active Node.js Version: v16.3.0

No problems detected.

Find help at https://github.com/coreybutler/nvm-windows/wiki/Common-Issues
```

**Comment by Marck36** [comment]
Hi, i got stuck in the same situation, spending some hours on the problem i find out that when using the command 'nvm use <version>' on my machine (that is a company's computer) it don't give me the privileges needed to do this task.
The problem is: on MVN version 1.1.11 when you run the 'use' command the feedback is that it successfully changed to the node version you specified but when running the 'node -v' command it isn't recognized, so i changed to MVN version 1.1.9 to see if was a version problem, then following the same steps it came to the same thing, but now it gives a better feedback, it says you don't have the privileges to do the path change.
Now knowing the root of the problem what i did was write a ps1 script that changes the path of the nodejs 'NVM_SYMLINK' on windows path variables after i change the node version by MVN.
![WhatsApp Image 2023-06-30 at 11 22 43](https://github.com/coreybutler/nvm-windows/assets/90698578/9092e79c-3d29-4cc8-914d-e5228c0f03f5)
![WhatsApp Image 2023-06-30 at 09 31 02](https://github.com/coreybutler/nvm-windows/assets/90698578/a19e27e5-ad51-4659-8539-218165c5113a)

**Comment by Marck36** [comment]
```
param(
    [string]$version
)

# Define o caminho da variável NVM_SYMLINK
$nvmdir = "$env:APPDATA\nvm\v$version"

# Define a variável de ambiente NVM_SYMLINK
[Environment]::SetEnvironmentVariable("NVM_SYMLINK", $nvmdir, "User")

# Exibe a confirmação
Write-Host "###### NVM_SYMLINK foi configurada com sucesso." -ForegroundColor Green
Write-Host "# .... Version: $version" -ForegroundColor Yellow
Write-Host "# .... Path: $nvmdir" -ForegroundColor Cyan

Import-Module $env:ChocolateyInstall\helpers\chocolateyProfile.psm1
refreshenv

# How use:
# RUN: nvm install 12.22.12
# RUN: .\nvm.ps1 -version 12.22.12
# RUN: nvm use 12.22.12
# RUN: node -v
```

**Comment by ykfq** [comment, marked as answer]
Found the right way to setup nvm-noinstall package without the `NVM_SYMLINK` trouble:
- unzip `nvm-noinstall.zip` to anywhere, eg: `C:\Users\bob\Apps\nvm`
- open powershell and run `C:\Users\bob\Apps\nvm\install.cmd`
- paste `C:\Users\bob\Apps\nvm` after the propmt
- open powershell or cmd and run `nvm install 22 lts` to install the `LTS` version of nodejs 22

Note, if you are none admin user, you shoul change: 
 - edit `install.cmd` and change `setx /M` to `setx`
 - edit `install.cmd` and change `set NVM_SYMLINK=C:\Program Files\nodejs` to `set NVM_SYMLINK=%NVM_PATH%\nodejs`
 - `nvm use` will not work, run `mklink /J nodejs v20.19.5` in cmd in the `C:\Users\bob\Apps\nvm` directory
